staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -172,14 +172,18 @@ type LeaderElector struct {
 func (le *LeaderElector) Run(ctx context.Context) {
 	defer func() {
 		runtime.HandleCrash()
-		le.config.Callbacks.OnStoppedLeading()
+		if le.config.Callbacks.OnStoppedLeading != nil {
```

Comment:
This really shouldn't be optional. There's no point to this client if they are. If a leader ignores OnStoppedLeading, it's going to carry on as if it's fine. The default behavior should be that the process quits. If a leader ignores OnStartedLeading, then what was the point of it joining an election group in the first place?

Reply:
`OnStoppedLeading` callback could be optional. In this case it certainly is at least. You are right about `OnStartedLeading` though, added for the sake of completion, but I can remove that check.

Reply:
> `OnStoppedLeading` callback could be optional.

Why? The whole idea is that you have one thing exclusively doing work. For example: leader A has networking problems, it loses leadership. B steps up. Networking problems are fixed, but A ignored the OnStoppedLeading notification. Now but A and B are doing things.

Reply:
In this specific case, leader election was used to only let one kubeadm client modify the kubeadm-config config map at a given time. When one started leading it would proceed with the modification. Then, the lock would be released when the leader election was cancelled on the leader. There's literally no action or logic to run when an instance stopped leading.

However, and as mentioned, I think they despite this was hinted to be used with leader election we can just Update the config map and in case of a conflict, get the config map again, mutate locally and try to update again.

I will open a PR with the alternative that does not involve leader election as mentioned, so no changes will be required at all on the leader election code.

Reply:
I see your point and that's certainly wrong on this PR, will update if we prefer to go with leader election.

Reply:
Thanks. I'm primarily worried about extending this API in ways that enable incorrect usage.